microsoftgraph/microsoft-graph-explorer-v4
# Fix: Set consent-type property

## Overview
Fixes #2439 

### Demo
![image](https://user-images.githubusercontent.com/45680252/227127365-2f56b5d6-9da1-4f5a-90bf-0557c2836263.png)
A spinner becomes available when the consent-type property is being fetched

When the fetch fails, a retry button is available to re-fetch the consent-type property
![image](https://user-images.githubusercontent.com/45680252/227186779-26bca53b-9439-45f5-a8ad-073e4c65a16e.png)


### Notes
The API call that returns the grants of the tenant takes time to return an updated grant after granting consent. For this reason, we have to poll the endpoint until we have an update. 

**The identity endpoints read data from eventually consistent datastores. This means that after updating a field, it takes time for all nodes to achieve the same state(eventual consistency). This means that calling the endpoint for updated data might still return stale data even after a write. 
For this reason, we have to keep polling the endpoint until we get the response that we expect. This is why we have a loop that continuously calls the grant endpoint until we have an updated grant.**

## Testing Instructions

* How to test this PR
Sign in with your account
Notice that all consent-type properties are available for consented permissions
Consent to a permission and notice the spinner just before the consent-type property is updated
Click on the retry button to re-fetch the consent-type property
